routers/web/repo/compare.go

```diff
@@ -215,21 +215,25 @@ func ParseCompareInfo(ctx *context.Context) *CompareInfo {
 	)
 
 	infoPath = ctx.Params("*")
-	infos := strings.SplitN(infoPath, "...", 2)
-
-	if len(infos) != 2 {
-		infos = []string{baseRepo.DefaultBranch, infoPath}
-		if strings.Contains(infoPath, "..") {
-			infos = strings.SplitN(infoPath, "..", 2)
-			ci.DirectComparison = true
-			ctx.Data["PageIsComparePull"] = false
+	var infos []string
+	if infoPath == "" {
+		infos = []string{baseRepo.DefaultBranch, baseRepo.DefaultBranch}
+	} else {
+		infos = strings.SplitN(infoPath, "...", 2)
+		if len(infos) != 2 {
+			infos = []string{baseRepo.DefaultBranch, infoPath}
+			if strings.Contains(infoPath, "..") {
+				infos = strings.SplitN(infoPath, "..", 2)
+				ci.DirectComparison = true
+				ctx.Data["PageIsComparePull"] = false
+			}
 		}
-	}
 
-	if len(infos) != 2 {
-		log.Trace("ParseCompareInfo[%d]: not enough compared branches information %s", baseRepo.ID, infos)
-		ctx.NotFound("CompareAndPullRequest", nil)
-		return nil
+		if len(infos) != 2 {
+			log.Trace("ParseCompareInfo[%d]: not enough compared branches information %s", baseRepo.ID, infos)
```

Comment:
It's impossible to get `len(infos) != 2` now, if yes, then there should be a bug. Should we use `log.Critical` here?